File: Algorithm/src/BaekJoon11568.java
import java.util.Scanner;

public class BaekJoon11568 {

	private static Scanner sc;

	public static void main(String[] args) {

		sc = new Scanner(System.in);

		int N = sc.nextInt();
		int[] arr = new int[N];
		int[] dp = new int[N];
		dp[0] = 1;
		for (int i = 0; i < N; i++) {
			arr[i] = sc.nextInt();
		}
		for (int i = 1; i < N; i++) {
			dp[i] = 1;
			for (int j = 0; j < i; j++) {
				if (arr[i] > arr[j]) {
					dp[i] = Math.max(dp[i], dp[j] + 1);
				}
			}
		}
		int max = 0;
		for (int i = 0; i < N; i++) {
			max = Math.max(max, dp[i]);
		}
		System.out.println(max);
	}
}

File: Algorithm/src/BaekJoon11568.kt
import java.util.Scanner

fun main(args: Array<String>) {

	val sc: Scanner = Scanner(System.`in`);

	val N: Int = sc.nextInt();
	var arr: IntArray = IntArray(N);
	var dp: IntArray = IntArray(N);
	dp[0] = 1;
	for (i in 0 until N) {
		arr[i] = sc.nextInt();
	}
	for (i in 1 until N) {
		dp[i] = 1;
		for (j in 0 until i) {
			if (arr[i] > arr[j]) {
				dp[i] = Math.max(dp[i], dp[j] + 1);
			}
		}
	}

	var max: Int = 0;
	for (i in 0 until N) {
		max = Math.max(max, dp[i]);
	}
	println(max);
}